pkgs/os-specific/darwin/apple-source-releases/libffi/package.nix

```diff
@@ -70,6 +70,12 @@ mkAppleDerivation (finalAttrs: {
       # Use `macCatalyst` instead of `iosmac` to avoid errors due to invalid availability annotations.
       substitute darwin/include/ffi.h "$dev/include/ffi.h" --replace-fail iosmac macCatalyst
     ''
+    # Match Apple’s configuration, which enables the legacy closure API by default on x86_64-darwin
+    + ''
+      substituteInPlace "$dev/include/ffitarget.h" \
+        --replace-fail '#define FFI_CLOSURES 1' '#define FFI_CLOSURES 1
+      #define FFI_LEGACY_CLOSURE_API ${if stdenv.hostPlatform.isx86_64 then "1" else "0"}'
+    ''
```

Comment:
Perhaps we should just install `darwin/include` wholesale rather than (or in addition to) the upstream headers? It has an `ffitarget.h` that conditionally includes the header for the various supported platforms and sets things appropriately. That way we don’t have to worry about keeping in sync; I’d worry about other differences in the configuration given that we’ve already had to replace one header and patch another.

Reply:
I opted to copy the platform-specific headers into the target config folder created by the `configure` script. That makes sure they’re consistent with the build, and lets the standard install phase handle copying them into place.